src/test/java/org/fogbowcloud/manager/core/ApplicationFacadeTest.java

```diff
@@ -12,1701 +12,2209 @@
+        } catch (UnauthorizedRequestException e) {
+            // verify
+            Assert.assertEquals(OrderState.OPEN, order.getOrderState());
+        }
+    }
 
-		Mockito.doReturn(order.getFederationUser()).when(this.aaaController).getFederationUser(Mockito.anyString());
+    // test case: When calling the createCompute() method the Order passed by parameter must return
+    // set to Open OrderState after its activation.
+    @Test
+    public void testCreateComputeOrder() throws Exception {
 
-		Mockito.doNothing().when(this.aaaController).authorize(Mockito.any(FederationUser.class),
-				Mockito.any(Operation.class), Mockito.any(Order.class));
+        // set up
+        ComputeOrder order = createComputeOrder();
+        Assert.assertNull(order.getOrderState());
 
-		Assert.assertNull(order.getOrderState());
+        Mockito.doReturn(order.getFederationUser()).when(this.aaaController)
+                .getFederationUser(Mockito.anyString());
 
-		String federationTokenValue = "";
-		this.application.createCompute(order, federationTokenValue);
+        Mockito.doNothing().when(this.aaaController).authenticate(Mockito.anyString());
 
-		Assert.assertEquals(OrderState.OPEN, order.getOrderState());
-	}
+        Mockito.doNothing().when(this.aaaController).authorize(Mockito.any(FederationUser.class),
+                Mockito.any(Operation.class), Mockito.any(Order.class));
 
-	@Test
-	public void testCreateComputeOrderUnauthenticated() throws Exception {
-		ComputeOrder order = createComputeOrder();
+        // exercise
+        this.application.createCompute(order, FEDERATION_TOKEN_VALUE);
 
-		Mockito.doThrow(new UnauthenticatedUserException()).when(this.aaaController).authenticate(Mockito.anyString());
+        // verify
+        Assert.assertEquals(OrderState.OPEN, order.getOrderState());
+    }
 
-		Mockito.doReturn(order.getFederationUser()).when(this.aaaController).getFederationUser(Mockito.anyString());
+    // test case: When try calling the createCompute() method without
+    // authentication, it must throw UnauthenticatedUserException.
+    @Test
+    public void testCreateComputeOrderUnauthenticated() throws Exception {
 
-		Mockito.doNothing().when(this.aaaController).authorize(Mockito.any(FederationUser.class), Mockito.any(Operation.class),
-				Mockito.any(Order.class));
+        // set up
+        ComputeOrder order = createComputeOrder();
 
-		try {
-			this.application.createCompute(order, "");
-			Assert.fail();
-		} catch (UnauthenticatedUserException e) {
-			Assert.assertNull(order.getOrderState());
-		}
-	}
+        Mockito.doThrow(new UnauthenticatedUserException()).when(this.aaaController)
+                .authenticate(Mockito.anyString());
 
-	@Test
-	public void testCreateComputeOrderTokenUnauthenticated() throws Exception {
-		ComputeOrder order = createComputeOrder();
+        Mockito.doReturn(order.getFederationUser()).when(this.aaaController)
+                .getFederationUser(Mockito.anyString());
```

Comment:
I think there is something wrong in the mock setup. The point is: how can this order.getFederationUser be called if we are going to throw an exception earlier? This is a symptom we are missing verifies. Also the description of the test case is not code, see the comment I made before.

Reply:
I will try to modify.